archiver.go

```diff
@@ -400,10 +391,8 @@ func (a *Archiver) pushChangesToRootedRepositories(
 		cancel()
 	}
 
-	if len(changes) == 0 {
-		if err := a.Store.UpdateFetched(r, *now); err != nil {
-			logger.Errorf(err, "error updating repository in database")
-		}
+	if err := a.Store.UpdateFetched(r, *now); err != nil {
+		logger.Errorf(err, "error updating repository in database")
```

Comment:
I guess now there's no need to append to `failedInits` all the ics here but we should append something if there is an error so `checkFailedinits` return an error in the return statement (or probably better, just directly return here an error).

On the other hand, reviewing the `failedInits` variable I've realized that it's modified right away from the go routines.